### PR TITLE
refactor: use & for block preprocessing

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

**Comment by matklad** [comment]
Follow up to https://github.com/near/nearcore/pull/6889

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -4367,7 +4367,7 @@ impl<'a> ChainUpdate<'a> {
     /// to process the block an the block is valid.
     //  Note that this function does NOT introduce any changes to chain state.
     fn preprocess_block(
-        &mut self,
+        &self,
```
Comment (matklad) [inline chain/chain/src/chain.rs:4370]: cc @mzhangmzz I think you'll like this! 

The fact that we only need `&` means that we can move the whole thing to chain, where it seems it belongs!

**Comment by matklad** [comment]
Blocked on https://github.com/near/nearcore/pull/6895

**Review by mzhangmzz — APPROVED** [review]

**Comment by matklad** [comment]
@mzhangmzz are you planning to move more logic from ChainUpdate to chain? If you do, it's probably best to close this PR then

**Comment by matklad** [comment]
closing in favor of https://github.com/near/nearcore/pull/6962

**Pull request closed without merging**